internal/backend/testing.go

```diff
@@ -24,7 +24,14 @@ import (
 
 // TestBackendConfig validates and configures the backend with the
 // given configuration.
+// Deprecated: Use TestTypedBackendConfig instead.
 func TestBackendConfig(t *testing.T, b Backend, c hcl.Body) Backend {
```

Comment:
What's the point of making this stuff generic? it feels like it's not utilized too much. I think we could just keep the concrete typing and chalk it up to YAGNI